federated_aws_cli/cache.py

```diff
@@ -80,6 +103,31 @@ def _safe_write(path):
     f.close()
 
 
+def write_aws_cli_credentials(credentials, role_arn, role_map):
+    # such as infosec-somerole
+    profile = _role_to_profile_name(role_arn, role_map)
+
+    # We call aws a bunch of times, getting all the return values
+    retval = 0
+
+    # Update the values
+    for cred_key, aws_key in viewitems(CREDENTIALS_TO_AWS_MAP):
+        if cred_key in credentials:
+            process = ["aws", "configure", "set",
+                       aws_key, credentials[cred_key],
+                       "--profile", profile]
+
+            _retval = subprocess.call(process)
```

Comment:
I looked at how to do this by calling the awscli module directly within python (not launching a subprocess) but gave up around here

```
from awscli.customizations.configure import get as configure_get
import StringIO
import argparse

output = StringIO.StringIO()
command = configure_get.ConfigureGetCommand(None, stream=output)
result = command('mfa_serial', argparse.Namespace())
```